inginious/frontend/plugins/multilang/static/hdlgrader.js

```diff
@@ -0,0 +1,458 @@
+        output = line;
+      }
+
+      //result.push(output);
+  }
+  result.push('{name:"Inputs:"},');
+  let start = 65; // ASCII for 'A'
+  for(var signal in inputsC){
+    if (signal in noBinary){
+        if (inputsC[signal] === inputsG[signal] && dataC[signal] === dataG[signal] ){
+            result.push('{name:"'+signal+'", wave: "'+inputsG[signal]+'", data:['+dataG[signal]+']},');
+        }else{
+            dataDiff = parseDiffWaveDromBus(inputsC[signal], inputsG[signal], dataC[signal], dataG[signal]);
+            result.push('{name:"'+signal+'", '+dataDiff+'},');
+            result.push('{name:"'+signal+'*", wave: "'+inputsG[signal]+'", data:['+dataG[signal]+']},');
+        }
+    }else{
+        if ( inputsC[signal] === inputsG[signal]){
+            result.push('{name:"'+signal+'", wave: "'+inputsC[signal]+'"},');
+        }else{
+            nodes = parseDiffWaveDrom(inputsC[signal], inputsG[signal], start);
+            result.push('{name:"'+signal+'", wave: "'+inputsC[signal]+'", node:"'+nodes+'"},');
+            result.push('{name:"'+signal+'*", wave: "'+inputsG[signal]+'"},');
+            start += (nodes.length - (nodes.match(/\./g) || []).length);
+        }
+    }
+  }
+  result.push('{name:"Outputs:"},');
+  for(var signal in outputsC){
+    if (signal in noBinary){
+        if (outputsC[signal] === outputsG[signal] && JSON.stringify(dataC[signal]) == JSON.stringify(dataG[signal])  ){
+            result.push('{name:"'+signal+'", wave: "'+outputsG[signal]+'", data:['+dataG[signal]+']},');
+        }else{
+            dataDiff = parseDiffWaveDromBus(outputsC[signal], outputsG[signal], dataC[signal], dataG[signal]);
+            result.push('{name:"'+signal+'", '+dataDiff+'},');
+            result.push('{name:"'+signal+'*", wave: "'+outputsG[signal]+'", data:['+dataG[signal]+']},');
+        }
+    }else{
+        if ( outputsC[signal] === outputsG[signal]){
+            result.push('{name:"'+signal+'", wave: "'+outputsG[signal]+'"},');
+        }else{
+            nodes = parseDiffWaveDrom(outputsC[signal], outputsG[signal], start);
+            result.push('{name:"'+signal+'", wave: "'+outputsC[signal]+'", node:"'+nodes+'"},');
+            result.push('{name:"'+signal+'*", wave: "'+outputsG[signal]+'"},');
+            start += (nodes.length - (nodes.match(/\./g) || []).length);
+        }
+    }
+  }
+  result.push("]");
+
+  if (start != 65){
+    result.push(",\"edge\":[");
+    edges = []
+    for(let j = 65; j<start;j+=2)
+        edges.push("'"+String.fromCharCode(j)+"|-|"+String.fromCharCode(j+1)+" diff'")
+    result.push(edges.join(", "));
+    result.push("]");
+  }
+  result.push("}");
+  result.push('</script>');
+  result.push('<script type="text/javascript">function loadWaveDrom(){ var x = document.getElementById("loadButtonWD"); WaveDrom.ProcessAll(); x.style.display = "none";}</script>');
+  result.push('<button id="loadButtonWD" onclick="loadWaveDrom()">Load</button> ');
```

Comment:
Is it still necessary to use the button?

Reply:
When the scripts are loaded in the init.py file, this button is not necessary. A call to WaveDrom.ProcessAll() is performed after update the html block.